ansible_base/authentication/utils/claims.py

```diff
@@ -150,7 +150,7 @@ def _add_rbac_role_mapping(has_permission, role_mapping, role, organization=None
 
 def process_groups(trigger_condition: dict, groups: list, authenticator_id: int) -> Optional[bool]:
```

Comment:
I like where you are taking this to return None or True and then letting `revoke` convert None to False. 
```suggestion
def process_groups(trigger_condition: dict, groups: list, authenticator_id: int) -> Tuple[None, True]:
```

Reply:
Regarding PyCharm, this is not a valid type hint - there can't be value, ony type, that's why I didn't update it

Reply:
Oh interesting, yea, because True is a value itself.... but its funny because its really not a bool. What about using Literal (updated suggestion).

Reply:
good one! yeah, that's what I missed